src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java

```diff
@@ -628,22 +627,22 @@ private static int calculateTagColumn(Matcher javadocTagMatcher,
      */
     private static List<JavadocTag> getMultilineNoArgTags(final Matcher noargMultilineStart,
             final String[] lines, final int lineIndex, final int tagLine) {
-        final String param1 = noargMultilineStart.group(1);
-        final int col = noargMultilineStart.start(1) - 1;
         final List<JavadocTag> tags = new ArrayList<>();
-        int remIndex = lineIndex + 1;
-        while (remIndex < lines.length) {
-            final Matcher multilineCont = MATCH_JAVADOC_MULTILINE_CONT
-                    .matcher(lines[remIndex]);
-            if (multilineCont.find()) {
-                remIndex = lines.length;
-                final String lFin = multilineCont.group(1);
-                if (!lFin.equals(NEXT_TAG)
-                    && !lFin.equals(END_JAVADOC)) {
-                    tags.add(new JavadocTag(tagLine, col, param1));
-                }
-            }
+        int remIndex = lineIndex;
+        Matcher multilineCont;
+
+        do {
```

Comment:
The original while loop condition could never be hit, we will always find the end.

Originally, this and `getMultilineArgTags` was very similar code. We could keep the other method the same without changing it because it could accept a javadoc that was only 1 line long and could try to go beyound the lines limit.
This one can't do the same because of the regular expression used to match javadocs.
> `^\s*(?>\*|\/\*\*)?\s*@(return|see)\s*$`

Because of the `$` on the end, this expression guarantees that there will be another line in the javadoc and doesn't end with `*/`. Another expression and method is used if it does end with `*/`, so we can never reach the limit of the lines here.
If this was not the case, we would eventually get an out of index exception and regression shows none.